azure/servicemanagement/__init__.py

```diff
@@ -1353,13 +1353,16 @@ def __getitem__(self, index):
 
 class DataVirtualHardDisk(WindowsAzureData):
 
-    def __init__(self):
-        self.host_caching = None
-        self.disk_label = u''
-        self.disk_name = u''
-        self.lun = 0
-        self.logical_disk_size_in_gb = 0
-        self.media_link = u''
+    def __init__(self, media_link=None, source_media_link=None,
+                 host_caching=None, disk_label=None, disk_name=None,
+                 lun=None, logical_disk_size_in_gb=None):
+        self.host_caching = host_caching
+        self.disk_label = disk_label
+        self.disk_name = disk_name
+        self.lun = lun
```

Comment:
changing the default value of `lun` and `logical_disk_size_in_gb` from 0 to None is a breaking change.  the automatic xml deserialization knows to parse the xml as an integer if the default value is an integer, otherwise it will keep it as a string.  For example, when you query a virtual hard disk, you'll get a lun '1' instead of 1.  I know, it's a bad situation, and we are moving away from that kind of automatic xml deserialization for the next gen library.